dart/lib/src/transport/transport.dart

```diff
@@ -1,9 +1,11 @@
 import 'dart:async';
 
+import '../sentry_envelope.dart';
 import '../protocol.dart';
 
-/// A transport is in charge of sending the event either via http
+/// A transport is in charge of sending the event/envelope either via http
 /// or caching in the disk.
 abstract class Transport {
-  Future<SentryId> send(SentryEvent event);
+  Future<SentryId?> sendSentryEvent(SentryEvent event);
+  Future<SentryId?> sendSentryEnvelope(SentryEnvelope envelope);
```

Comment:
ideally, we'd have `send(SentryEnvelope envelope)` only but for testing would require the deserialization of the envelope